api/src/pcapi/scripts/clean_offer_titles_with_eans/main.py

```diff
@@ -0,0 +1,242 @@
+        try:
+            func(offer_rows)
+        except Exception:  # pylint: disable=broad-exception-caught
+            retry_one_chunk_at_a_time(offer_rows)
+            return False
+        return True
+
+    return inner
+
+
+@atomic()
+@retry_and_log
+def update_unknown_offers(offer_rows: Collection[OfferEanQueryRow]) -> None:
+    reject_offers(offer_rows)
+
+
+@atomic()
+@retry_and_log
+def update_gcu_incompatible_offers(offer_rows: Collection[OfferEanQueryRow]) -> None:
+    reject_offers(offer_rows)
+
+
+@atomic()
+@retry_and_log
+def update_legit_offers(offer_rows: Collection[OfferEanQueryRow]) -> None:
+    ids = {row.id for row in offer_rows}
+    legit_offers = Offer.query.filter(Offer.id.in_(ids))
+
+    offer_to_product = {row.id: row for row in offer_rows}
+
+    with atomic():
+        for offer in legit_offers:
+            offer.name = offer_to_product[offer.id].product_name
+
+            if offer_to_product[offer.id].product_json_data:
+                extra_data = cast(str, offer_to_product[offer.id].product_json_data)
+                offer.extraData = json.loads(extra_data)
+            db.session.add(offer)
+
+
+def reject_offers(offer_rows: Collection[OfferEanQueryRow]) -> None:
+    def cancel_booking(offer: Offer) -> None:
+        cancelled_bookings = bookings_api.cancel_bookings_from_rejected_offer(offer)
+        for booking in cancelled_bookings:
+            on_commit(
+                functools.partial(
+                    transactional_mails.send_booking_cancellation_by_pro_to_beneficiary_email,
+                    booking,
+                    rejected_by_fraud_action=True,
+                )
+            )
+
+    def notify_offerer(offer: Offer) -> None:
+        if offer.venue.bookingEmail:
+            recipients = [offer.venue.bookingEmail]
+        else:
+            recipients = [recipient.user.email for recipient in offer.venue.managingOfferer.UserOfferers]
+
+        offer_data = transactional_mails.get_email_data_from_offer(
+            offer, offer.validation, OfferValidationStatus.REJECTED
+        )
+        on_commit(
+            functools.partial(
+                transactional_mails.send_offer_validation_status_update_email,
+                offer_data,
+                recipients,
+            )
+        )
+
+    ids = {row.id for row in offer_rows}
+    base_query = Offer.query.filter(
+        Offer.id.in_(ids),
+        Offer.status != OfferValidationStatus.REJECTED.value,
+    )
+
+    for offer in base_query:
+        cancel_booking(offer)
+        notify_offerer(offer)
+
+    base_query.update(
+        {
+            "validation": OfferValidationStatus.REJECTED.value,
+            "lastValidationDate": datetime.now(tz.utc),  # pylint: disable=datetime-now
+            "lastValidationType": OfferValidationType.AUTO.value,
+            "lastValidationAuthorUserId": None,
+            "isActive": False,
+        },
+        synchronize_session=False,
+    )
```

Comment:
Pour que le script soit opérationnel, il manque ici : 
```suggestion
    )
if __name__ == "__main__":
    app.app_context().push()
    run()
```